src/test/java/org/cactoos/io/JoinedTest.java

```diff
@@ -52,4 +53,24 @@ public void joinsOk() throws IOException {
             new TextHasString("firstsecondthird")
         );
     }
+
+    /**
+     * Must join inputs of the iterable in the given order.
```

Comment:
@ilyakharlamov What's is the final result of a joined bettween files? A file with all contents? Please, add a test to check it.

Reply:
@fabriciofx `Joined` works only with an abstract interface `Input`. `Input` can be anything - a sequence of bytes from memory, a  stream of bytes from the network or any other sequence, depending on the implementation. So probably if we need a test, the test should go to a concrete implementation like ResourceOfTest of something. `Joined` class must have nothing to do with files, isn't it?

Reply:
@ilyakharlamov The problem is we have a `Joined` class which works with all kind of `Input`s, but we're testing only with strings. IMHO, it should be tested with others types of `Input`s, like a file, to check the expected result is right. WDYT?

Reply:
@fabriciofx It's nice to have lots of tests, but why files if the interface has nothing to do with the files? I just don't understand what's so special about the files that cannot be replaced by another InputStream.

Reply:
@ilyakharlamov I said "Files" just for example. But you can use another `Input` different of strings?

Reply:
@fabriciofx these classes depend on the interfaces, not the implementation. Same for their tests.

Reply:
@llorllale Sorry, but I didn't get the point. Can you explain it better?

Reply:
@llorllale I didn't get it. Can you explain better?

Reply:
@fabriciofx what @ilyakharlamov said [here](https://github.com/yegor256/cactoos/pull/1038#discussion_r250641704)

Reply:
@llorllale Right, but if `Joined` works with `Input`s why can't test it with others inputs different of text? And seems reasonable test it in `JoinedTest`.

Reply:
@fabriciofx this PR already tests `Joined` against the `Input` interface. If any specific implementation of `Input` is misbehaving, feel free to open a ticket for it. Or, if `Joined` is failing with a specific combination of `Input`s, feel free to open a ticket about it.

Reply:
@llorllale Right.

@ilyakharlamov Can you open an issue for it?